### PR TITLE
Add Circle config and basic README

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#6211

We need to add the Kong API gateway as a managed app to our incubator catalog.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+workflows:
+  package-and-push-chart-on-tag:
+    jobs:
+      - architect/push-to-app-catalog:
+          name: "package and push kong-app chart"
+          app_catalog: "giantswarm-incubator-catalog"
+          app_catalog_test: "giantswarm-incubator-test-catalog"
+          chart: "kong-app"
+          # Trigger job on git tag.
```
Comment (tomahawk28) [inline .circleci/config.yml]:
```suggestion
          # Needed to trigger job also on git tag.
```
The same sentence as `aws-operator`, `chart-operator`

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+version: 2.1
+orbs:
+  architect: giantswarm/architect@0.1.2
+
+workflows:
+  package-and-push-chart-on-tag:
```
Comment (tomahawk28) [inline .circleci/config.yml]: Just curious, Don't we need to push on docker first?

Comment (rossf7) [reply .circleci/config.yml]: I don't think so as there is no image. The chart will use retagged versions of the Kong images.

File: .circleci/config.yml (hunk 1 above)
Comment (rossf7) [reply .circleci/config.yml]: Thanks, I took this from prometheus-operator-app I'll update there too.

**Comment by rossf7** [comment]
Done in #4.

**Pull request closed without merging**